### PR TITLE
feat: rename trafficMesh flag to edgeHandlers

### PR DESCRIPTION
**Pull request by tiffanosaurus**
**- Summary**
The `--trafficMesh` flag is not a super clear term to users, let's go for `--edgeHandlers` which is more to the point. Deprecate the current flag in favor of a. new flag and add a warning to users that use the old one.

image

**- Test plan**

Choose your poison:
- `npm run test` (follow Contributing.md for instructions)
- `netlify dev --edgeHandlers`
- `netlify dev -eh`

**- Description for the changelog**

Rename `netlify dev` flag `--trafficMesh` (short: `-t`) to `--edgeHandlers` (short: `-eh`) and update description to `activates the edge handlers runtime`.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by tiffanosaurus** [comment]
I have this as `fix:` but maybe `chore:` is better? It is a breaking change even if it's hidden

**Comment by erquhart** [comment]
Thanks @tiffanosaurus!

cc/ @erezrokah I believe this is okay if `--trafficMesh` is considered beta, but defer to you on the decision.

File: src/commands/dev/index.js
Comment (erezrokah) [inline src/commands/dev/index.js:317]: I believe this should be `Edge Handlers` per our style guide cc @rstavchansky

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @tiffanosaurus is there any issue we can discuss this?
To my knowledge the `trafficMesh` flag does more than add [Edge Handlers](https://www.netlify.com/products/edge/edge-handlers/) support (which allows running custom logic at the edge).
Should we should call it [`edge`](https://www.netlify.com/products/edge/) or `local-edge` or something similar.

Also, could we keep the old flag and add a warning message to users instructing them to switch to the new flag so they can self serve?
We can drop the old flag completely after a couple of weeks with the new one.

As for the PR type:
1. `chore:` is a change that isn't visible to users (doesn't even require a new release)
2. `fix:` is a bug fix (suggests a patch version)
3. `feat:` is a user facing change (suggests a minor version) - so I believe this should be a `feat:`

To declare a breaking change you can use `feat!:`, but this isn't once since:
1. The flag is hidden (like you mentioned).
2. This is considered Beta (again like you mentioned).

**Comment by tiffanosaurus** [comment]
Thanks @erezrokah! The issue and discussion around how to handle the flag can be found here: https://github.com/netlify/traffic-mesh/issues/1517

We can definitively add a new flag instead, we figured since it was beta, we could be more aggressive with the approach and replace it.

File: src/commands/dev/index.js
Comment (tiffanosaurus) [inline src/commands/dev/index.js:322]: Let me know if a different copy would be better here!

File: src/commands/dev/index.js
Comment (tiffanosaurus) [inline src/commands/dev/index.js:228]: Let me know if a different copy here would be better!

**Comment by tiffanosaurus** [comment]
Updated the PR with the new flag and a deprecation notice for the `--trafficMesh`.

This might have more changes as discussed in the issue since this not only enables edge handlers, but also the new redirect engine and people might want to enable it without wanting to use edge handlers.

File: src/commands/dev/index.js
Comment (mraerino) [inline src/commands/dev/index.js:321]: it's common in the unix world to only have single-letter flags following a single-dash parameter so that you can combine multiple single letters in one param, e.g. `tar -xzf` which is short for `tar -x -z -f`. this kinda breaks with this convention and might leave users confused if there are isolated `-e` or `-h` (help?) flags?

multi-letter flags should always use double-dashes as a prefix imho

**Comment by erezrokah** [comment]
Thank you for making these changes @tiffanosaurus. Can we finalize the discussion [here](https://github.com/netlify/traffic-mesh/issues/1517) before merging this? Also great point by @mraerino about the single letter convention.

**Comment by tiffanosaurus** [comment]
Yep of course and will update for the letter convention

**Comment by rstavchansky** [comment]
Heads up that the `--trafficMesh flag` is currently publicly documented. https://docs.netlify.com/routing/edge-handlers/configure-and-deploy/#local-development
@jacklyn-net for a docs update

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
@tiffanosaurus when you merge this can you keep the commit message as `feat: rename trafficMesh flag to edgeHandlers (#1906)` and clear the commit description from any intermediate messages? 
That will ensure an automated version bump as a minor version.

**Pull request merged**
